In [54]:
import pandas as pd
import numpy as np
import math

### 1.
Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

всего производится n=100 выстрелов, вероятность попадания при каждом p=0,8, вероятность промаха q=1−p=1−0,8=0,2. Нужно найти, что будет ровно k= 85 попаданий. Подставляем все в формулу Pn(k)=Ckn⋅pk⋅(1−p)n−k=Ckn⋅pk⋅qn−k

In [ ]:
P = P100(85 )= C85 100 * 0.8**85 * 0.2**15 = 0.0481

### 2. 
Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Вероятность, что лампочка исправна:

In [2]:
1 - 0.0004 = 0.9996

Какова вероятность, что ни одна из них не перегорит в первый день?

In [22]:
P = 0.9996 ** 5000 = 0.13528114551440706

 Какова вероятность, что перегорят ровно две?

In [78]:
Р1 = 0.0004 * 0.0004 = 1.6e-07

### 3. 
Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [74]:
p = 0.5

n = 144

q = 1-p

k = 70

C_nk = math.factorial(n) / (math.factorial(k)*(math.factorial(n-k)))
C_70 = C_nk * p**k * q**(n-k)
print(f'Вероятность, что монета, подброшенная 144 раза выпадет орлом ровно 70 раз = {round(C_70, 4)}')

Вероятность, что монета, подброшенная 144 раза выпадет орлом ровно 70 раз = 0.0628


In [37]:
n = 2**144

In [40]:
Сnk = C144 70 = 144!/70!(144-70)!

In [56]:
m = math.factorial(144)/(math.factorial(70)*math.factorial(74))

In [42]:
P = m/n

In [46]:
P

0.06281178035144776

In [ ]:
Р1 (70,144)= С (70,144)*(1/2)**70 *(1/2)**74

In [43]:
d = (1/2)**70* (1/2)**74

In [44]:
P1 = m * d

In [45]:
P1

0.06281178035144776

Ответ: 0.0628

### 4. 
В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

In [ ]:
C2 10 = 10!/2!(10-2)! = 10! / 2! * 8! = 45

In [ ]:
P(2) = 7/10 * 6/9 = 0.467

In [ ]:
C2 11 = 11!/2!(11-2)! = 11! / 2! * 9! = 55

In [ ]:
P(2) = 9/11 * 8/10 = 0.655

In [ ]:
P = 0.467 * 0.655 = 0.305885

Какова вероятность того, что ровно два мяча белые?

In [ ]:
Р(А) = 7/10 = 0.7

In [ ]:
Р(В) = 9/11 = 0.82

In [ ]:
Р(АВ) =Р(А)*P(B)= 0.7 * 0.82 = 0.574

Какова вероятность того, что хотя бы один мяч белый?

Найдем вероятность противоположного события: вытащим все черные.

In [ ]:
р= 3/10 * 2/11 = 0.3 * 0.19 = 0.057

Тогда вероятность, что хотя бы один мяч белый:

In [ ]:
1-р = 1-0.057 = 0.943